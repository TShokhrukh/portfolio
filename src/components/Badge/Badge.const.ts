import { cn } from '@bem-react/classname'

export const cnBadge = cn('Badge')
