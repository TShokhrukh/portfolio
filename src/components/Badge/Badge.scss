
.Badge {
  padding: 5px 10px;
  border-radius: $border-radius;
  font-weight: bold;
}