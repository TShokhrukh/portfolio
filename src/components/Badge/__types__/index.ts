import type { ReactNode, ElementType } from 'react'
import type { IClassNameProps } from '@bem-react/core'

export interface IBadgeProps extends IClassNameProps {
  role?: string,
  id?: string,
  component?: ElementType,
  children?: ReactNode
}

export interface IBadgeColorProps {
  color?: 'purple'
}
