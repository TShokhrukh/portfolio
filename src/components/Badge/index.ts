export * from './build'
export * from './__types__'
export { Badge as BadgeBase } from './Badge'
