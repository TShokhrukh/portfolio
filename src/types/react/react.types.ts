import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
