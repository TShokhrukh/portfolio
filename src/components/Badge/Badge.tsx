import React from 'react'
import { cnBadge } from './Badge.const'
import type { IBadgeProps } from './__types__'

import './Badge.scss'

export const Badge: React.FC<IBadgeProps> = ({
  component: Component = 'span',
  className,
  children,
  ...props
}) => {
  return (
    <Component {...props} className={cnBadge({}, [className])}
      >{children}</Component>
  )
}
