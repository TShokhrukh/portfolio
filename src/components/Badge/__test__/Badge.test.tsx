/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { Badge } from '../Badge'

describe('Badge', () => {
  test('component template', () => {
    render(<Badge role='badge' id='temp_id' className='temp_class' />)

    expect(screen.getByRole('badge').tagName).toMatchSnapshot()
  })

  test('render the children', () => {
    render(<Badge data-testid='badge'>children</Badge>)

    expect(screen.getByTestId('badge')).toHaveTextContent('children')
  })

  test('must render SPAN tag', () => {
    render(<Badge data-testid='badge' />)

    expect(screen.getByTestId('badge').tagName).toBe('SPAN')
  })
})
