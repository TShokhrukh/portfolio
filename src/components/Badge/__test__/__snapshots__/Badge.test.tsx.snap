// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component template 1`] = `"SPAN"`;
